### PR TITLE
[MISC] Rename gap_decorator::unaligned_seq_type to gap_decorator::unaligned_sequence_type.

### PR DESCRIPTION
**Pull request by simonsasse**
Resolves: https://github.com/seqan/product_backlog/issues/322

### CODE REVIEW

File: include/seqan3/range/decorator/gap_decorator.hpp [hunk 1]
```diff
@@ -137,7 +137,7 @@ class gap_decorator
      *
      * \stableapi{Since version 3.1.}
      */
-    using unaligned_seq_type = inner_type;
+    using unaligned_sequence_type = inner_type;
```
Comment (eseiler) [inline include/seqan3/range/decorator/gap_decorator.hpp:140]: You need to also add a deprecated alias, i.e.

```cpp
//!\deprecated Please use seqan3::gap_decorator::unaligned_sequence_type instead.
using unaligned_seq_type SEQAN3_DEPRECATED_310 = unaligned_sequence_type;
```

Not sure if you need the doxygen comment, you can check if it actually does somthing

Comment (simonsasse) [reply include/seqan3/range/decorator/gap_decorator.hpp:140]: I'll add it. thx. :)
How should I check if the doxygen comment is needed ?

Comment (eseiler) [reply include/seqan3/range/decorator/gap_decorator.hpp:140]: I would build it and check if the alias is shown (https://seqan3-git-fork-simonsasse-gapdecoratorrename-seqan.vercel.app/usr/html/classseqan3_1_1gap__decorator.html#a441d8562fac08376653af3325c4326f8)

If it's shown, it should have the `deprecated` comment

Comment (simonsasse) [reply include/seqan3/range/decorator/gap_decorator.hpp:140]: It was neither in the user nor the dev documentation so I removed the oxygen comment. Then the documentation test from the CI failed because the member unaligned_seq_type was not documented. So I now added it back in. Let's see :)

**Review by Irallia — APPROVED** [review]
LGFM, thank you!

File: include/seqan3/range/decorator/gap_decorator.hpp [hunk 2]
```diff
@@ -321,7 +324,7 @@ class gap_decorator
         return iterator{*this, pos1};
     }
 
-    /*!\brief Assigns a new sequence of type seqan3::gap_decorator::unaligned_seq_type to the decorator.
+    /*!\brief Assigns a new sequence of type seqan3::gap_decorator::unaligned_sequence_type to the decorator.
      * \param[in,out] dec       The decorator to modify.
      * \param[in]     unaligned The unaligned sequence to assign.
      * \details
```
Comment (eseiler) [inline include/seqan3/range/decorator/gap_decorator.hpp:330]: A few lines below there is
`template <typename unaligned_seq_t>`

since this is public, can you change it to `template <typename unaligned_sequence_t>` to reflect the general naming scheme?

Can you check if there are other occurrences in the gap_decorator?

File: test/unit/alignment/aligned_sequence_test_template.hpp [hunk 3]
```diff
@@ -36,16 +36,16 @@ TYPED_TEST_P(aligned_sequence, fulfills_concept)
 
 TYPED_TEST_P(aligned_sequence, assign_unaligned_sequence)
 {
```
Comment (eseiler) [inline test/unit/alignment/aligned_sequence_test_template.hpp:38]: The changes in the file are not related to `gap_decorator::unaligned_seq_type` - just revert them

File: include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp [hunk 4]
```diff
@@ -64,14 +64,14 @@ struct unaligned_seq<t>
...
 
 // customisation point for our gap decorators.
-//!\brief Exposes the unaligned sequence type if *t* exposes the type member `unaligned_seq_type`.
+//!\brief Exposes the unaligned sequence type if *t* exposes the type member `unaligned_sequence_type`.
 template <typename t>
 //!\cond
-    requires requires { typename std::remove_reference_t<t>::unaligned_seq_type; }
+    requires requires { typename std::remove_reference_t<t>::unaligned_sequence_type; }
 //!\endcond
 struct unaligned_seq<t>
```
Comment (eseiler) [inline include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp:72]: the name of this type trait can stay as is because it is detail (in contrast to the `template <typename unaligned_seq_t>` in gap_decorator)

**Review by eseiler — CHANGES_REQUESTED** [review]

File: include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp (hunk 4 above)
Comment (simonsasse) [reply include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp:72]: ok that makes sense. thx :)

**Review by eseiler — APPROVED** [review]
